src/main/java/de/tum/cit/aet/artemis/fileupload/domain/FileUpload.java

```diff
@@ -0,0 +1,113 @@
+package de.tum.cit.aet.artemis.fileupload.domain;
+
+import java.time.ZonedDateTime;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.Table;
+
+import de.tum.cit.aet.artemis.core.domain.DomainObject;
+import de.tum.cit.aet.artemis.fileupload.service.FileUploadScheduleService;
+
+@Entity
+@Table(name = "file_upload")
+public class FileUpload extends DomainObject {
+
+    @Column(name = "path", nullable = false)
+    private String path;
+
+    @Column(name = "server_file_path", nullable = false)
+    private String serverFilePath;
+
+    @Column(name = "filename", nullable = false)
+    private String filename;
+
+    /**
+     * This represents the id of the entity connected to this FileUpload.
+     * Nullable in case file uploads happen before the creation of
+     * the corresponding entity. Warning: If left null, the FileUpload
+     * will be cleaned up after a timeframe specified in
+     * {{@link FileUploadScheduleService}}.
+     */
+    @Column(name = "entity_id")
+    private Long entityId;
+
+    /**
+     * This represents the type of the entity that is connected to this
+     * FileUpload. The same as above applies to this value.
+     */
+    @Enumerated
+    @Column(name = "entity_type")
+    private FileUploadEntityType entityType;
+
+    @Column(name = "created_at", nullable = false)
+    private ZonedDateTime createdAt;
```

Comment:
I'd rename this to `creation_date`

Reply:
Good call, that way its more consistent with the other models. I changed it in this [commit](https://github.com/ls1intum/Artemis/pull/9970/commits/bc743c81d0316ce0b6edec100dd1f9fab9d11a88). 👍